### PR TITLE
QueryWithIrrelevant usage example and showcase

### PR DESCRIPTION
**Pull request by DoctorVoid**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@enigmatis/mongo-driver": "file:../mongo-driver",
     "@enigmatis/polaris": "file:../polaris",
     "@enigmatis/polaris-logs": "^2.5.0",
-    "@enigmatis/utills": "^1.2.0",
+    "@enigmatis/utills": "file:../utills",
```
Comment (DoctorVoid) [inline package.json]: will change when other pull requests merge

**Review by osher-sade — APPROVED** [review]

File: src/schema/resolvers/book-resolvers-functions.ts [hunk 2]
```diff
@@ -46,6 +48,24 @@ export const bookQueryResolver = async (
...
+    parent: object | null,
+    query: any,
+    context: PolarisContext,
+) => {
+    const { realityId, dataVersion } = context.headers;
+    if (!Number.isInteger(realityId as any)) {
+        throw new UserInputError('please provide reality-id header');
+    } else {
+        return QueryWithIrrelevant(
+            BookModelPerReality(context),
```
Comment (bikov) [inline src/schema/resolvers/book-resolvers-functions.ts]: BookModelPerReality(context) is creating new schema / finding the schema in mongoose.models, please call it once and save it as variable

**Review by bikov — APPROVED** [review]

File: src/schema/resolvers/book-resolvers.ts [hunk 3]
```diff
@@ -12,6 +13,7 @@ export const resolvers = {
     },
     Query: {
         books: bookQueryResolver,
+        booksStartWith: bookStartWithQueryResolver,
```
Comment (yarinvak) [inline src/schema/resolvers/book-resolvers.ts]: starts

File: src/schema/resolvers/book-resolvers-functions.ts [hunk 4]
```diff
@@ -46,6 +48,25 @@ export const bookQueryResolver = async (
         return BookModelPerReality(context).find({});
     }
 };
+export const bookStartWithQueryResolver = async (
```
Comment (yarinvak) [inline src/schema/resolvers/book-resolvers-functions.ts]: starts

**Pull request merged**